src/Controller/FilesController.php

```diff
@@ -78,16 +81,30 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        if (false === $user instanceof FrontendUser) {
+            throw new PageNotFoundException();
+        }
```

Comment:
This will now never show the login page (i.e. 401 page), please remove it again.

Reply:
and what goes there instead? `throw new InsufficientAuthenticationException();`?

Reply:
No, the implementation needs to be different in general. The algorithm needs to do the following:

Prior to your PR:

* If a protected folder is accessed with no groups defined: respond with 404.
* If a protected folder is accessed with groups defined and no user is logged in: respond with 401.
* If a protected folder is accessed with groups defined and a user is logged in, but does not have access: respond with 403.
* If a protected folder is accessed with groups defined and a user is logged in and does have access: return the file.

New feature:

* If a protected folder is accessed with no groups defined and it is not a home directory of any user: respond with 404.
* If a protected folder is accessed with groups defined or it is a home directory of a user and no user is logged in: respond with 401.
* If a protected folder is accessed with groups or it is a home directory of a user defined and a user is logged in, but the user does not have access to the folder (either group wise or home directory wise): respond with 403.
* If a protected folder is accessed with groups defined and a user is logged in and the user does have access (either group wise or home directory wise): return the file.

The implementation needs to reflect that. I can implement it when I get around to it and can update your PR accordingly.

Reply:
o.k.